mirrord/config/src/feature/network/outgoing.rs

```diff
@@ -1,17 +1,75 @@
+use core::str::FromStr;
+use std::net::SocketAddr;
+
 use mirrord_analytics::CollectAnalytics;
 use mirrord_config_derive::MirrordConfig;
 use schemars::JsonSchema;
+use serde::Deserialize;
+use thiserror::Error;
 
 use crate::{
     config::{from_env::FromEnv, source::MirrordConfigSource, ConfigError},
     util::{MirrordToggleableConfig, VecOrSingle},
 };
 
+/// List of addresses/ports/subnets that should be sent through either the remote pod or local app,
+/// depending how you set this up with either `remote` or `local`.
+///
+/// You may use this option to specify when outgoing traffic is sent from the remote pod (which
+/// is the default behavior when you enable outgoing traffic), or from the local app (default when
+/// you have outgoing traffic disabled).
+///
+/// Takes a list of values, such as:
+///
+/// - Only UDP traffic on subnet `1.1.1.0/24` on port 1337 will go throuh the remote pod.
+///
+/// ```json
+/// {
+///   "remote": ["udp://1.1.1.0/24:1337"]
+/// }
+///
+/// - Only UDP and TCP traffic on resolved address of `google.com` on port `1337` and `7331`
+/// will go through the remote pod.
+/// ```json
+/// {
+///   "remote": ["google.com:1337", "google.com:7331"]
+/// }
+/// ```
+/// 
+/// - Only TCP traffic on `localhost` on port 1337 will go through the local app, the rest will
+///   be emmited remotely in the cluster.
+/// ```json
+/// {
+///   "local": ["tcp://localhost:1337"]
+/// }
+///
+/// - Only outgoing traffic on port `1337` and `7331` will go through the local app.
+/// ```json
+/// {
+///   "local": [":1337", ":7331"]
+/// }
+/// ```
+///
+/// Valid values follow this pattern: `[protocol]://[name|address|subnet/mask]:[port]`.
+#[derive(Deserialize, PartialEq, Eq, Clone, Debug, JsonSchema)]
+#[serde(untagged, rename_all = "lowercase")]
+pub enum OutgoingFilterConfig {
+    /// Traffic that matches what's specified here will go through the remote pod, everything else
+    /// will go through local.
+    Remote(VecOrSingle<String>),
+
+    /// Traffic that matches what's specified here will go through the local app, everything else
+    /// will go through the remote pod.
+    Local(VecOrSingle<String>),
```

Comment:
This config uses `VecOrSingle`, which I think expects a semicolon-delimited string (`"stuff;more stuff"`), but in all of the documentation, example and tests, you use lists (`["stuff", "more stuff"]`). That probably doesn't work, right? Maybe check what the parsed configuration ends up being with those configs?
If as I suspect `VecOrSingle` does not parse lists, then either change this to something that does (probably just `Vec`?) or change the docs, examples and tests - whatever's quicker to do.

Reply:
Had to remove the `untagged` attribute, now the schema recognizes it as `filter: anyOf(object, object)` (something like that), which is what we wanted, where the `object`s are `remote` or `local`.

The list works with `,`, and I think it also supports custom separators (when you want to concat into string).